Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -144,9 +144,13 @@ bool loadCanceled
     }
 
     public async Task<DafnyDocument> VerifyAsync(DafnyDocument document, CancellationToken cancellationToken) {
-      var request = new VerifyRequest(document, cancellationToken);
-      requestQueue.Add(request, cancellationToken);
-      return await request.Document.Task;
+      try {
+        var request = new VerifyRequest(document, cancellationToken);
+        requestQueue.Add(request, cancellationToken);
+        return await request.Document.Task;
+      } catch (OperationCanceledException) {
```

Comment:
Personally I like having the `try` around as little code as possible, to make it clear where it's trying to catch. So if you have the try/catch in this file, I would put it around the line `var verificationResult = verifier.Verify(document.Program, cancellationToken);`

However, @camrein preferred not handling `OperationCanceledException` in this file at all, so you could move the try/catch to a `VerifyAsyncHandleCancellation` in `DocumentDatabase`

Reply:
Indeed, I'd prefer to leave the cancellation logic centralized within `DocumentDatabase`.
Therefore, I propose adding a `private` method inside `DocumentDatabase` that takes responsibility for the verification cancellation. This method can then be used by `VerifyDocumentIfRequiredAsync` and `LoadAndVerifyAsync` in place of `documentLoader.VerifyAsync`.

Reply:
That makes a lot of sense. I'll do that along with a deeper investigation of the false reports of verification success described above.

Reply:
That refactoring is now applied, along with a fix to avoid discarding non-consecutive document updates (which are allowed by the protocol, and can show up when saving documents, but were previously rejected by the server).

It's not a completely bug-free experience yet. The document can get into an inconsistent state where the parser or name resolver think there's an error and can't be convinced otherwise. But that's better than incorrectly reporting verification success. Reloading the document always seems to fix the issue.